src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -616,6 +616,67 @@ def handleCondorStatusService(self):
 
         return
 
+    def resizeCores(self, numCores):
+        """
+        _resizeCores_
+
+        Look at the job runtime environment and determine whether we are allowed
+        to resize the core count.  If so, change the numCores request to be the
+        number of CPUs in this batch slot.
+
+        This only works when running under HTCondor, $_CONDOR_MACHINE_AD exists,
+        and:
+          - Cpus is a positive integer.
+          - WMCore_ResizeJob is 'true'
+        Note that we do not implement the ClassAd language - hence no expressions.
+
+        Returns the resized number of cores; if resizing is disabled, returns
+        the value of the `numCores` input parameter.
+        """
+        machineAdFile = os.environ.get('_CONDOR_MACHINE_AD')
+        jobAdFile = os.environ.get('_CONDOR_JOB_AD')
+        if not machineAdFile or not jobAdFile:
+            return numCores
+        resizeJob = None
+        machineCpus = None
+        try:
+            with open(jobAdFile) as fd:
+                for line in fd:
+                    line = line.strip()
+                    info = line.split("=", 1)
+                    if len(info) != 2:
+                        continue
+                    attr = info[0].strip().lower()  # Need case-insensitive compares.
+                    val = info[1].strip()
+                    if attr == 'wmcore_resizejob':
```

Comment:
HTCondor ClassAds are not case sensitive, hence we can't use a case-sensitive compare here.